format/src/main/scala/scala/meta/internal/fmt/SyntacticGroup.scala

```diff
@@ -50,14 +52,8 @@ object SyntacticGroup {
     Literal.precedence == Term.SimpleExpr1.precedence &&
       Literal.precedence == Pat.SimplePattern.precedence
   )
-  case object Path extends Type with Term with Pat {
+  case class Path(tree: Tree) extends Type with Term with Pat {
```

Comment:
Would it be possible to avoid this tree field by handling the needs parentheses case in ApplyUnary? I'm a bit concerned about taxing all usages of Path when we could handle the numeric literal case at the root.

Reply:
You prefer to have all the parenthesis logic in TreePrinter ?

Reply:
No 😉 but in this particular case for unary apply on numeric literal however it seems custom handling in TreePrinter beats introducing complexity into syntactic groups, which is used in a lot of places. It's cleaner to see in TreePrinter that unary apply requires custom handling, then make syntactic groups look more complicated than they already are.